src/H5Fint.c

```diff
@@ -1850,56 +1860,54 @@ H5F_open(const char *name, unsigned flags, hid_t fcpl_id, hid_t fapl_id)
      * application's responsibility to prevent this situation (there's no
      * way for us to detect it here anyway).
      */
-    if (drvr->cmp)
+    if (drvr->cmp) {
         tent_flags = flags & ~(H5F_ACC_CREAT | H5F_ACC_TRUNC | H5F_ACC_EXCL);
-    else
-        tent_flags = flags;
 
-    /*
-     * When performing a tentative open of a file where we have stripped away
-     * flags such as H5F_ACC_CREAT from the specified file access flags, the
-     * H5E_BEGIN/END_TRY macros are used to suppress error output since there
-     * is an expectation that the tentative open might fail. Even though we
-     * explicitly clear the error stack after such a failure, the underlying
-     * file driver might maintain its own error stack and choose whether to
-     * display errors based on whether the library has disabled error reporting.
-     * Since we wish to suppress that error output as well for the case of
-     * tentative file opens, surrounding the file open call with the
-     * H5E_BEGIN/END_TRY macros is an explicit instruction to the file driver
-     * not to display errors. If the tentative file open call fails, another
-     * attempt at opening the file will be made without error output being
-     * suppressed.
-     *
-     * However, if stripping away the H5F_ACC_CREAT flag and others left us
-     * with the same file access flags as before, then we will skip this
-     * tentative file open and only make a single attempt at opening the file.
-     * In this case, we don't want to suppress error output since the underlying
-     * file driver might provide more details on why the file open failed.
-     */
-    if (tent_flags != flags) {
-        /* Make tentative attempt to open file */
-        H5E_BEGIN_TRY
```

Comment:
As an example, I believe switching from `H5E_BEGIN_TRY` -> `H5E_PAUSE_ERRORS` (inside `H5FD_try_open`) here will cause some VOL connectors/VFDs to start printing out error stacks from failures during speculative file opens because, at the time, the easiest way to determine whether or not to print an error stack was to check if the library's default error stack has error reporting disabled. Since I setup these connectors/drivers with their own error stack separate from the library, it generally only makes sense for that plugin to print out the contents of its own error stack on the way out from a "top-level" function (e.g., one of the callback functions the plugin implements). The error macros I setup check whether the library's default error stack has a callback function set before deciding whether to print out the error stack on the way out. With these changes, it seems I'd instead need to know whether error reporting on the error stack has been paused.

Reply:
OK, that makes sense.

I can add an API routine (H5Eis_paused ?) that will allow plugin developers to query if the library has paused error reporting on the default stack.  I don't think it's a good idea to let those developers unpause the default stack, but I could also add a H5Epause & H5Eresume for them to have similar behavior for their own stacks.

I'll try to get these additional routines added tomorrow.

Reply:
Added new routines